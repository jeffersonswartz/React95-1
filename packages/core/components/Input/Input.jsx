import styled from "styled-components";
import { padding, borders, shadow } from "styled-system";

const Input = styled.input`
  outline: none;
  border: none;
  cursor: text;

  ${padding}
  ${borders}
  ${shadow}
`;

Input.defaultProps = {
  pt: 3,
  pr: 3,
  pb: 5,
  pl: 3,
  borderBottom: 0,
  borderRight: 0,
  borderTopWidth: 1,
  borderTopStyle: 1,
  borderTopColor: "grays.3",
  borderLeftWidth: 1,
  borderLeftStyle: 1,
  borderLeftColor: "grays.3",
  boxShadow: "input"
};

export default Input;
