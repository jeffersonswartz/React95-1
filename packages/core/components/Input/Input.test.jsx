import React from 'react';
import { render } from '../shared/test/utils';
import Input from './Input';

describe('<Input />', () => {
  describe('Snapshots', () => {
    it('should match snapshot', () => {
      const { container } = render(<Input />);
      expect(container).toMatchSnapshot();
    });
  });
});
