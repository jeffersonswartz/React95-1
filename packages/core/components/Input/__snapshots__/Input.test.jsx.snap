// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> Snapshots should match snapshot 1`] = `
.c0 {
  outline: none;
  border: none;
  cursor: text;
  padding-top: 3px;
  padding-right: 3px;
  padding-bottom: 5px;
  padding-left: 3px;
  border-bottom: 0;
  border-right: 0;
  border-top-width: 1px;
  border-top-style: solid;
  border-top-color: #868a8e;
  border-left-width: 1px;
  border-left-style: solid;
  border-left-color: #868a8e;
  box-shadow: inset -1px -1px 0 0 #c3c7cb, inset 1px 1px 0 0 #000000, 0.5px 0.5px 0 0.5px #ffffff;
}

<div>
  <input
    class="c0"
  />
</div>
`;
